commands/FBWhyCommands.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python
+# Example file with custom commands, located at /magical/commands/example.py
+
+import lldb
+import fblldbbase as fb
+import fblldbviewhelpers as viewHelpers
+import fblldbobjecthelpers as objcHelpers
+
+def lldbcommands():
+  return [ 
+    FBPrintWhyViewNotVisible(),
+    FBPrintWhyViewNotInteractable(),
+  ]
+
+class FBPrintWhyViewNotVisible(fb.FBCommand):
+  def name(self):
+    return 'wnvisible'
+
+  def description(self):
+    return 'Print the reasons the given view is not visible.'
+
+  def args(self):
+    return [ fb.FBCommandArgument(arg='aView', type='UIView/NSView *', help='The view to check why it isn\'t visible.') ]
+
+  def run(self, arguments, options):
+    view = arguments[0]
+
+    if not viewHelpers.isView(view):
+      print 'Argument is not a view\n'
+      return
+
+    reasons = ''
+
+    # width or height is 0
+    size = '((CGRect)[(id)' + view + ' frame]).size'
+    width = float(fb.evaluateExpression(size + '.width'))
+    height = float(fb.evaluateExpression(size + '.height'))
+    if width == 0.0 or height == 0.0:
+      reasons += 'The width or height is 0. Did you forget to set the frame?\n'
+
+    # view is out of bounds of its superview
+    origin = '((CGRect)[(id)' + view + ' frame]).origin'
+    xValue = float(fb.evaluateExpression(origin + '.x'))
+    yValue = float(fb.evaluateExpression(origin + '.y'))
+    if xValue < 0.0 or yValue < 0.0:
```

Comment:
This will produce false positives. Negative origin isn't necessarily a cause of views not showing.

Could this use `CGRectIntersectsRect` to see if a view's frame shares space with its superview?